README.md

```diff
@@ -16,7 +18,86 @@ cd gryffin
 pip install .
 ```
 
-### Reference 
+## Running Gryffin
+
+Gryffin identifies optimal parameter choices in a closed-loop approach. Given a set of prior observations, consisting of evaluated parameters and associated measurements, it can recommend several parameter choices for future evaluation.
+
+A basic skeleton of the code to run Gryffin is as follows:
+```
+from gryffin import Gryffin
+
+##  BUDGET (int): number of evaluations of the black-box function we want to optimize.
+##  benchmark (function): 'black box' function that returns the merit obtained for a certain parameter setup.
+
+# create an instance from a configuration file
+CONFIG_FILE = 'config.json'
+gryffin = Gryffin(CONFIG_FILE)
+
+observations = []
+for _ in range(BUDGET):
+
+    # gryffin recommends a number of parameter setups to evaluate based on previous observations
+    samples  = gryffin.recommend(observations = observations)
+   
+    for sample in samples:
+        # black box models is queried with the recommended parameter setups 
+        measurement   = benchmark(sample)
+        sample['obj'] = measurement
+        observations.append(sample)
+```
+
+## Configuration file
+
+The config file (`config.json`) defines the optimization problem and the setup of Gryffin. It must contain the three following fields: `general`, `parameters` and `objectives`.
+
+- `general`. It contains hyperparameter choices that will affect the performance of Gryffin. The whole list of hyperparameters and their default values are available at `src/gryffin/utilities/defaults.py`. Some of the most important hyperparameters (type) are:
+    - `auto_desc_gen` (bool): It enables dynamic Gryffin for the case it is set to `True` and descriptors are provided.
+    - `sampling_strategies` (int): Number of acquisition functions as per the Phoenics framework. 
+
+- `parameters`. It characterizes the parameters that are to be optimized. Gryffin supports both continuous, categorical and discrete parameters. Each parameter type is characterized with a number of attributes.
+
+    - **Continuous** parameters are characterized by an upper (`high`) and lower (`low`) bound.
+    - **Categorical** parameters take on one of a limited number of possible options. Options may be characterized by descriptors. Both information should be available in `cat_details`.
+    - **Discrete** parameters also take on one of a limited number of possible options. Options are in the interval defined by an upper (`high`) and lower (`low`) bound.
```

Comment:
Are we still using the `stride` attribute for the discrete parameters? If so, I would recommend adding something like the following "... and lower (`low`) bound, separated by a interval `stride`."

Reply:
no stride defined in the code... Dont really know how this part works, indeed.

lets keep it in mind just in case  we have to get back to this in the future.